ckanext/datastore/backend/postgres.py

```diff
@@ -1555,8 +1555,17 @@ def search_sql(context, data_dict):
         context['connection'].execute(
             u'SET LOCAL statement_timeout TO {0}'.format(timeout))
 
-        get_names = datastore_helpers.get_table_and_function_names_from_sql
-        table_names, function_names = get_names(context, sql)
+        function_names = datastore_helpers.get_function_names_from_sql(context, sql)
+        for f in function_names:
+            for name_variant in [f.lower(), '"{}"'.format(f)]:
+                if name_variant in backend.allowed_sql_functions:
+                    break
+            else:
+                raise toolkit.NotAuthorized({
+                    'permissions': ['Not authorized to call function {}'.format(f)]
+                })
```

Comment:
I see this logic already used in the lines below. Does this need to be duplicated or should one be removed?
https://github.com/OpenGov-OpenData/ckan/blob/ff034c374376d7fecadaf74f36887ac4785334fa/ckanext/datastore/backend/postgres.py#L1579-L1587

Reply:
removed